File: test-suite/src/test/java/io/micronaut/docs/jdbc/refresh/DbSecretStore.java
package io.micronaut.docs.jdbc.refresh;

import jakarta.inject.Singleton;

import java.util.concurrent.atomic.AtomicReference;

/**
 * Stands in for an external secret service (Vault, a cloud secret manager, ...) in the documentation example.
 */
@Singleton
public class DbSecretStore {

    private final AtomicReference<String> password = new AtomicReference<>();

    public String currentPassword() {
        return password.get();
    }

    public void rotate(String newPassword) {
        password.set(newPassword);
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/jdbc/refresh/DbSecretStore.kt
package io.micronaut.docs.jdbc.refresh

import jakarta.inject.Singleton
import java.util.concurrent.atomic.AtomicReference

/**
 * Stands in for an external secret service (Vault, a cloud secret manager, ...) in the documentation example.
 */
@Singleton
class DbSecretStore {

    private val password = AtomicReference<String>()

    fun currentPassword(): String? = password.get()

    fun rotate(newPassword: String) = password.set(newPassword)
}
